llmfoundry/models/layers/norm.py

```diff
@@ -111,9 +112,65 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
                             self.eps).to(dtype=x.dtype)
 
 
+class TritonRMSNorm(torch.nn.Module):
+
+    def __init__(
+        self,
+        normalized_shape: Union[int, List[int], torch.Size],
+        eps: float = 1e-5,
+        dropout_p: float = 0.0,
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[torch.device] = None,
+    ):
+        if isinstance(normalized_shape, int):
```

Comment:
how come the others don't need this if/else stuff?

Reply:
FA expects flat tensor since they only consider the NLP mode of LayerNorm, whereas torch's is more generic